File: relaax/templates/environments/basic/environment/bandit.py
import numpy as np


class Bandit(object):

    def __init__(self):
        # List out our bandits.
        # Default bandit 4 (index#3) is set to most often provide a positive reward.
        self.slots = [0.2, 0, -0.2, -5]

    def pull(self, action):
        result = np.random.randn(1)
        if result > self.slots[action]:
            return 1
        else:
            return -1

File: sample/basic-js/environment/src/app/bandit.js

function bandit() {
  // List out our bandits
  // Default bandit 4 (index#3) is set to most often provide a positive reward.
  this.slots = [0.2, 0.5, 0.8, 0.0]
}

bandit.prototype.pull = function(action) { 
  result = Math.random()
  if(result > this.slots[action])
    return 1
  else
    return -1
}

module.exports = bandit;
